WeatherMonitoringSystem.Tests/DomainTests/CoreTests/WeatherStationTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Moq;
+using FluentAssertions;
+using AutoFixture;
+using WeatherMonitoringSystem.Core;
+using WeatherMonitoringSystem.Bots;
+
+public class WeatherStationTests
+{
+    private readonly Fixture _fixture = new();
+    private readonly WeatherStation _weatherStation;
+
+    public WeatherStationTests()
+    {
+        _weatherStation = new WeatherStation();
+    }
+
+    [Fact]
+    public void AddBot_Should_Add_Bot_To_List()
+    {
+        var botMock = new Mock<IWeatherBot>();
+
+        _weatherStation.AddBot(botMock.Object);
+
+        _weatherStation.GetAllBots().Should().ContainSingle().Which.Should().Be(botMock.Object);
+    }
+
+    [Fact]
+    public void AddBot_Should_Throw_Exception_When_Bot_Is_Null()
```

Comment:
lets update all test names to follow this convention: `MethodName_StateUnderTest_ExpectedBehavior`.

i.e. `AddBot_ShouldThrowException_WhenBotIsNull`